draft-janfred-radext-radiusdtls-bis.md

```diff
@@ -183,26 +183,24 @@ While the client may be able to deduce the operational state of the local server
 
-When UDP is used as a transport protocol, the absence of a reply can cause a client to deduce (incorrectly) that the proxy is unavailable.
+The absence of a reply can cause a client to deduce (incorrectly) that the proxy is unavailable.
 The client could then fail over to another server or conclude that no "live" servers are available (OKAY state in {{!RFC3539}}, Appendix A).
 This situation is made even worse when requests are sent through a proxy to multiple destinations.
 Failures in one destination may result in service outages for other destinations, if the client erroneously believes that the proxy is unresponsive.
 
-For RADIUS/TLS, it is RECOMMENDED that implementations utilize the existence of a TCP connection along with the application-layer watchdog defined in {{RFC3539}}, Section 3.4 to determine that the server is "live".
-RADIUS/TLS clients MUST mark a connection DOWN, if the network stack indicates that the connection is no longer active.
-If the network stack indicates that the connection is still active, clients MUST NOT decide that it is down until the application-layer watchdog algorithm has marked it DOWN.
-RADIUS/TLS clients MUST NOT decide that a RADIUS/TLS server is unresponsive until all TLS connections to it have been marked down.[^contradiction]{:jf}
+It is REQUIRED that implementations utilize the existence of a TCP/DTLS connection along with the application-layer watchdog defined in {{RFC3539}}, Section 3.4 to determine that the server liveliness.
+RADIUS/TLS clients MUST mark a connection DOWN, under the following conditions:
+- The administrator has marked the connection administrative DOWN.
+- The network stack indicates that the connection is no longer viable.
+- The application-layer watchdog algorithm has marked it DOWN.
 
-[^contradiction]: The specification in RFC6613 is contradictory here. Section 2.4 says that it is recommended to have a watchdog. Section 2.6 says it must be used.
+For RADIUS/TLS, TCP connections MAY send TCP keepalives as described in {{RFC9293}} Section 3.8.4, that UDP connections MAY send periodic DTLS keepalives as defined in {{RFC6520}}, as a way of proactively and rapidly triggering a connection DOWN notification from the network stack.  These liveliness checks are essentially redundant in the presence of an application-layer watchdog, but may provide more rapid notifications of connectivity issues.
 
-The above requirements do not forbid the practice of a client proactively closing connections or marking a server as DOWN due to an administrative decision.
+It is REQUIRED that RADIUS/(D)TLS clients implement the Status-Server extension as described in {{?RFC5997}} as the application level watchdog to detect the liveness of the peer in the absence of responses.
```

Comment:
It's absolutely required that with proxy servers there is some active monitoring to determine upstream reachability.  Waiting for traffic to be sent to a proxy before marking it as down leads to a poor end user experience.

Reply:
Yes.